### PR TITLE
10739 - Multiple slashes at the end of route should not be accepted as valid url

### PR DESCRIPTION
**Pull request by andrewnester**
Fix for #10739

### CODE REVIEW

File: framework/web/UrlManager.php [hunk 1]
```diff
@@ -242,6 +242,12 @@ public function parseRequest($request)
 
             Yii::trace('No matching URL rules. Using default URL parsing logic.', __METHOD__);
 
+            // before processing pathInfo and suffix there should not be end multiple slashes in $pathInfo
+            $endSlashPattern = '/\/\/$/';
+            if (preg_match($endSlashPattern, $pathInfo)) {
```
Comment (SilverFire) [inline framework/web/UrlManager.php]: `substr_compare` should be faster

Comment (dynasource) [inline framework/web/UrlManager.php]: agreed. Wherever possible, preg_ functions should be avoided

Comment (andrewnester) [inline framework/web/UrlManager.php]: @dynasource @SilverFire fixed by using substr_compare instead

File: framework/web/UrlManager.php [hunk 2]
```diff
@@ -242,6 +242,12 @@ public function parseRequest($request)
 
             Yii::trace('No matching URL rules. Using default URL parsing logic.', __METHOD__);
 
+            // before processing pathInfo and suffix there should not be end multiple slashes in $pathInfo
+            $endSlashPattern = '//';
```
Comment (SilverFire) [inline framework/web/UrlManager.php]: Why do we need a separate variable for this? Comment above is enough to describe, what's going here

File: framework/web/UrlManager.php [hunk 3]
```diff
@@ -242,6 +242,12 @@ public function parseRequest($request)
 
             Yii::trace('No matching URL rules. Using default URL parsing logic.', __METHOD__);
 
+            // before processing pathInfo and suffix there should not be end multiple slashes in $pathInfo
```
Comment (SilverFire) [inline framework/web/UrlManager.php]: I'd like the following wording:

`Ensure, that $pathInfo does not end with more than one slash`

File: framework/web/UrlManager.php (hunk 2 above)
Comment (andrewnester) [inline framework/web/UrlManager.php]: Sounds reasonable to me

File: framework/web/UrlManager.php (hunk 3 above)
Comment (andrewnester) [inline framework/web/UrlManager.php]: yeah, this one sounds more clear

**Comment by SilverFire** [comment]
Thanx

**Comment by dynasource** [comment]
thanx

**Pull request merged**